pkg/scheduler/framework/types_test.go

```diff
@@ -1559,3 +1561,191 @@ func TestCalculatePodResourcesWithResize(t *testing.T) {
+				MatchLabelKeys:    []string{"bar"},
+			},
+			enableMatchLabelKeys: true,
+			want: &AffinityTerm{
+				TopologyKey:       "region",
+				Selector:          labels.SelectorFromSet(labels.Set{"bar": "a"}),
+				Namespaces:        sets.NewString("default"),
+				NamespaceSelector: labels.Everything(),
+			},
+		},
+		{
+			name: "matchLabelKeys is ignored when keys in matchlabelKeys doesn't exist in pod's label (feature flag: enabled)",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{}, // no labels
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAffinity: &v1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+								{
+									LabelSelector:     barLabelSelector,
+									TopologyKey:       "region",
+									Namespaces:        []string{"default"},
+									NamespaceSelector: &metav1.LabelSelector{},
+									MatchLabelKeys:    []string{"bar"},
+								},
+							},
+						},
+					},
+				},
+			},
+			term: &v1.PodAffinityTerm{
+				LabelSelector:     barLabelSelector,
+				TopologyKey:       "region",
+				Namespaces:        []string{"default"},
+				NamespaceSelector: &metav1.LabelSelector{},
+				MatchLabelKeys:    []string{"bar"},
+			},
+			enableMatchLabelKeys: false,
+			want: &AffinityTerm{
+				TopologyKey:       "region",
+				Selector:          barSelector, // no change in selector.
+				Namespaces:        sets.NewString("default"),
+				NamespaceSelector: labels.Everything(),
+			},
+		},
+		{
+			name: "feature flag disabled",
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"foo": "hoge", "bar": "a"},
+				},
+				Spec: v1.PodSpec{
+					Affinity: &v1.Affinity{
+						PodAffinity: &v1.PodAffinity{
+							RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+								{
+									LabelSelector:     barLabelSelector,
+									TopologyKey:       "region",
+									Namespaces:        []string{"default"},
+									NamespaceSelector: &metav1.LabelSelector{},
+									MatchLabelKeys:    []string{"bar"},
+								},
+							},
+						},
+					},
+				},
+			},
+			term: &v1.PodAffinityTerm{
+				LabelSelector:     barLabelSelector,
+				TopologyKey:       "region",
+				Namespaces:        []string{"default"},
+				NamespaceSelector: &metav1.LabelSelector{},
+				MatchLabelKeys:    []string{"bar"},
+			},
+			enableMatchLabelKeys: false,
+			want: &AffinityTerm{
+				TopologyKey:       "region",
+				Selector:          barSelector,
+				Namespaces:        sets.NewString("default"),
+				NamespaceSelector: labels.Everything(),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := newAffinityTerm(tt.pod, tt.term, tt.enableMatchLabelKeys)
+			if err != nil {
+				t.Fatalf("unexpected error is returned from newAffinityTerm: %v", err)
+			}
+			if d := cmp.Diff(tt.want, got); d != "" {
+				t.Fatalf("unexpected result from newAffinityTerm(): %s", d)
```

Comment:
there is a template for dumping the output like "+got, etc.", could you adapt to that pattern?